### ISSUE TITLE
DotConnect Identifiers 

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#13194]
I am seeing this in my logs:

```
WARN  2017-12-07 20:12:13,622 com.dotmarketing.filters.CMSUrlUtil:warn(181): XSS Found in Query String: b=${dotcmsReleaseVersion}
WARN  2017-12-07 20:12:13,630 com.dotmarketing.filters.CMSUrlUtil:warn(181): XSS Found in Query String: b=${dotcmsReleaseVersion}
ERROR 2017-12-07 20:12:15,144 com.dotmarketing.viewtools.BrowserAPI:velocityError(245): Could not load folders :  @ url:POST//demo.dotcms.com/dwr/call/plaincall/BrowserAjax.openFolderContent.dwr | lang:1 | ip:0:0:0:0:0:0:0:1 | Admin:true | start:12-07-2017 08:11:05 EST  ref:http://localhost:8080/c/portal/layout?p_l_id=b7ab5d3c-5ee0-4195-a17e-8f5579d718dd&p_p_id=site-browser&p_p_action=0&&dm_rlout=1&r=1512695496185&in_frame=true&frame=detailFrame&container=true
java.lang.NumberFormatException: For input string: ""
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65) ~[?:1.8.0_112]
	at java.lang.Long.parseLong(Long.java:601) ~[?:1.8.0_112]
	at java.lang.Long.parseLong(Long.java:631) ~[?:1.8.0_112]
	at java.text.DigitList.getLong(DigitList.java:195) ~[?:1.8.0_112]
	at java.text.DecimalFormat.parse(DecimalFormat.java:2051) ~[?:1.8.0_112]
	at java.text.SimpleDateFormat.subParse(SimpleDateFormat.java:1869) ~[?:1.8.0_112]
	at java.text.SimpleDateFormat.parse(SimpleDateFormat.java:1514) ~[?:1.8.0_112]
	at java.text.DateFormat.parse(DateFormat.java:364) ~[?:1.8.0_112]
	at com.dotmarketing.util.ConvertToPOJOUtil.getFolderFields(ConvertToPOJOUtil.java:297) ~[ConvertToPOJOUtil.class:?]
	at com.dotmarketing.util.ConvertToPOJOUtil.convertDotConnectMapToFolder(ConvertToPOJOUtil.java:261) ~[ConvertToPOJOUtil.class:?]
	at com.dotmarketing.portlets.folders.business.FolderFactoryImpl.getSubFolders(FolderFactoryImpl.java:173) ~[FolderFactoryImpl.class:?]
	at com.dotmarketing.portlets.folders.business.FolderFactoryImpl.getSubFoldersTitleSort(FolderFactoryImpl.java:125) ~[FolderFactoryImpl.class:?]
	at com.dotmarketing.portlets.folders.business.FolderFactoryImpl.getFoldersByParent(FolderFactoryImpl.java:306) ~[FolderFactoryImpl.class:?]
	at com.dotmarketing.portlets.folders.business.FolderAPIImpl.findSubFolders_aroundBody6(FolderAPIImpl.java:165) ~[FolderAPIImpl.class:?]
	at com.dotmarketing.portlets.folders.business.FolderAPIImpl$AjcClosure7.run(FolderAPIImpl.java:1) ~[FolderAPIImpl$AjcClosure7.class:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
```

**Comment by wezell** [comment]
Looks like we have a `static final DateFormat` in the ConvertToPOJOUtil.  I don't think it is thread safe.

**Comment by wezell** [comment]
Looks like there are a number of problems with the ConvertToPOJOUtil

1. We should be using the DotConnect.loadObjectResults() which gives us a map of java objects back, rather than just strings that need to be reparsed, which is error prone.

2. We don't need all the MultitreeFactory getChildrenClassByOrder methods. These need to go away.  I removed them all in my branch here, which probably should have been merged first:
https://github.com/dotCMS/core/tree/issue-13164-delete-template_container

3. Why are we using a static class ConvertToPOJOUtil?  Why not use a non static class that would be thread safe?  
`List<Identifier> list = new IdentifierTransformer(results).asList();`

4.  I think we should have small Transformer classes that are each responsible for transforming a single pojo or list of pojos, instead of one big class.
